shell/common/platform_view.cc

```diff
@@ -140,6 +140,33 @@ void PlatformView::HandlePlatformMessage(
     response->CompleteEmpty();
 }
 
+size_t PlatformView::RegisterTexture(std::shared_ptr<flow::Texture> texture) {
+  int texture_id;
+  fxl::AutoResetWaitableEvent latch;
```

Comment:
We use latches very sparingly. For example, during engine initialization where we are very sensitive to initialization order of various subsystems. So using latches here makes me quite nervous. You don't want the platform thread to block on all tasks on the GPU thread. Can you confirm that operation is performed once per texture? And, if so, document the same here please. Also, it would be clearer if you added the thread assertion macros here (for example`ASSERT_IS_PLATFORM_THREAD`). In general, I believe we should be doing a better job that we currently are about justifying the use of a latch.